gwnr/waveform/prepare_waveforms.py

```diff
@@ -0,0 +1,356 @@
+        input_insp_dat_rel_loc = Path(f"{self.sim_name}/Ecc{self.ecc}"
+                                 f"/Ev/Lev{self.lev}*/Run/"
+                                 "ApparentHorizons/Horizons.h5")
+        
+        input_rdown_dat_rel_loc = Path(f"{self.sim_name}/Ecc{self.ecc}"
+                                 f"/Ev/Lev{self.lev}_Ringdown/"
+                                 f"Lev{self.lev}*/Run/"
+                                 "ApparentHorizons/Horizons.h5")
+        
+        data_paths_insp = os.path.join(self.sim_dir, 
+                                  input_insp_dat_rel_loc)
+        
+        data_paths_rdown = os.path.join(self.sim_dir, 
+                                  input_rdown_dat_rel_loc)
+        if verbose:
+            run_cmd = "JoinH5 -v"
+            
+        else:
+            run_cmd = "JoinH5"
+            
+            
+        run_cmd += f" -o {self.joined_horizons_outfile_path}"\
+                            f" -l {data_paths_insp} {data_paths_rdown}"
+        
+        print(f"Running command\n {run_cmd}")
+        
+        
+        cmd = os.popen(run_cmd)
+    
+        out = cmd.read()
+    
+        print('Command output \n', out)
+        
+        
+        print("Command completed. Please check Errors.txt for details.")
+        
+    def transform_to_CoM_frame(self, 
+                               skip_beginning_fraction=0.01,
+                               skip_ending_fraction=0.10,
+                               file_format="NRAR",
+                               extrap_enn_list=[-1, 2, 3, 4, 5, 6]):
+        
+        from scri.SpEC.com_motion import remove_avg_com_motion
+        
+        print("Transforming to CoM frame...")
+        
+        for extrap_enn in extrap_enn_list:
+            
+            print(f"Working on Extrapolated N_{extrap_enn}")
+            
+            path_to_waveform_h5=os.path.join(self.extrap_out_dir, 
+                                f"rhOverM_Extrapolated_N{extrap_enn}.h5")
+
+            path_to_horizons_h5 = self.joined_horizons_outfile_path
+
+            remove_avg_com_motion(
+                              w_m=None,
+                              path_to_waveform_h5=path_to_waveform_h5,
+                              path_to_horizons_h5=path_to_horizons_h5,
+                              skip_beginning_fraction=skip_beginning_fraction,
+                              skip_ending_fraction=skip_ending_fraction,
+                              file_write_mode="w",
+                              m_A=None,
+                              m_B=None,
+                              file_format=file_format,
+                              write_corrected_file=True
+                              )
+            
+            
+    def upload_output_dir(self):
+        raise NotImplementedError
+        
+    def prepare_waveform(self, 
+                         verbose=False, 
+                         ChMass = 1.0,
+                         UseStupidNRARFormat = True, 
```

Comment:
stick to snake case, dont mix with camel case. --> `ch_mass` and `use_stupid_nrar_format`